lms/static/scripts/frontend_apps/utils/vitalsource.js

```diff
@@ -0,0 +1,14 @@
+/**
+ * A naive regex matcher for a VBID in a URL-like string.
+ *
+ * @param {string} url
+ * @returns {string|null} VBID, or `null` if nothing looks like a VBID in `url`
+ */
+export function bookIDFromURL(url) {
+  const bookIdPattern = /book\/([0-9A-Z-]+)(\/|$)/;
```

Comment:
The pattern should check for `/books/` rather than `/book/`. Example URL I just copied from VitalSource: https://bookshelf.vitalsource.com/#/books/9781400847402/cfi/6/8!/4/2@0:0

Reply:
Zoinks! I'm so glad you caught this.